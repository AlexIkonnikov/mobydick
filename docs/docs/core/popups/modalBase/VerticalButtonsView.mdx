import VerticalButtonsView from '../assets/VerticalButtonsView.png';

<img src={VerticalButtonsView} width="50%"/>

Reusable VerticalButtonsView component for ModalBase

## Example

```tsx
import {
  Button,
  createStyles,
  ModalBase,
  Typography,
  usePopups,
  useStyles,
  View,
} from '@lad-tech/mobydick-core';

export const ModalWidget = () => {
  const [styles] = useStyles(styleFn);
  const {openPopup} = usePopups();

  return (
    <View style={styles.container}>
      <Typography font={'Regular-Primary-H5'}>Modal</Typography>
      <Button
        text={'ModalBase'}
        onPress={() =>
          openPopup({
            Content: props => (
              <ModalBase {...props}>
                <ModalBase.VerticalButtonsView>
                  <ModalBase.VerticalButton text={'VerticalButton'} />
                </ModalBase.VerticalButtonsView>
              </ModalBase>
            ),
          })
        }
      />
    </View>
  );
};

const styleFn = createStyles(({spaces}) => ({
  container: {
    gap: spaces.Space8,
  },
}));
```


## `Props`

### [`View Props`](https://reactnative.dev/docs/textinput#props)

Inherits [View Props](https://reactnative.dev/docs/view#props)



