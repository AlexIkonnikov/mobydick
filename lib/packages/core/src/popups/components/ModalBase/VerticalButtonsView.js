"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const View_1 = __importDefault(require("../../../basic/components/View/View"));
const useStyles_1 = __importDefault(require("../../../styles/hooks/useStyles"));
const stylesCreate_1 = __importDefault(require("./stylesCreate"));
const VerticalButtonsView = props => {
    const [styles] = (0, useStyles_1.default)(stylesCreate_1.default);
    const { children } = props;
    return <View_1.default style={styles.verticalButtonsView}>{children}</View_1.default>;
};
exports.default = VerticalButtonsView;
