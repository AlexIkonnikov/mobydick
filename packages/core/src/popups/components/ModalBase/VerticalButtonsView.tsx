import {FC} from 'react';

import View from '../../../basic/components/View/View';
import useStyles from '../../../styles/hooks/useStyles';

import stylesCreate from './stylesCreate';

const VerticalButtonsView: FC<{
  children: React.ReactNode;
}> = props => {
  const [styles] = useStyles(stylesCreate);
  const {children} = props;

  return <View style={styles.verticalButtonsView}>{children}</View>;
};

export default VerticalButtonsView;
