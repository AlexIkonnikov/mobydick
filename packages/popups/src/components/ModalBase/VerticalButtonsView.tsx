import React, {FC} from 'react';
import {Button, ISize, ITypes} from '@npm/mobydick-cta';
import {useStyles} from '@npm/mobydick-styles';

import stylesCreate from './stylesCreate';

interface IFixedView {
  textOne: string;
  typeOne: ITypes;
  onPressOne(): void;

  textTwo?: string;
  typeTwo?: ITypes;
  onPressTwo?(): void;
}

const VerticalButtonsView: FC<IFixedView> = props => {
  const [styles] = useStyles(stylesCreate);
  const {textOne, typeOne, onPressOne, textTwo, typeTwo, onPressTwo} = props;

  const isShowTwoButton = textTwo && typeTwo && onPressTwo;
  return (
    <>
      <Button
        size={ISize.fixed}
        style={styles.verticalButtonOne}
        text={textOne}
        type={typeOne}
        onPress={onPressOne}
      />
      {isShowTwoButton && (
        <Button
          size={ISize.fixed}
          style={styles.verticalButtonTwo}
          text={textTwo}
          type={typeTwo}
          onPress={onPressTwo}
        />
      )}
    </>
  );
};

export default VerticalButtonsView;
