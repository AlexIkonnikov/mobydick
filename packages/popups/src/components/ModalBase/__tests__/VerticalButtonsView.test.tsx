import {render} from '@testing-library/react-native';
import {ITypes} from '@npm/mobydick-cta';
import React from 'react';

import VerticalButtonsView from '../VerticalButtonsView';

describe('@npm/mobydick-popups/modalBase', () => {
  it('should renders correctly VerticalButtonsView one button ', () => {
    const {toJSON} = render(
      <VerticalButtonsView
        typeOne={ITypes.destructive}
        onPressOne={() => console.log('onPress')}
        textOne={'text'}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
  it('should renders correctly VerticalButtonsView secondary ', () => {
    const {toJSON} = render(
      <VerticalButtonsView
        typeOne={ITypes.secondary}
        onPressOne={() => console.log('onPress')}
        textOne={'text'}
        typeTwo={ITypes.secondary}
        onPressTwo={() => console.log('onPress')}
        textTwo={'text'}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
