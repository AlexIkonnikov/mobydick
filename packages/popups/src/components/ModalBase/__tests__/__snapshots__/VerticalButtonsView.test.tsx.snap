// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@npm/mobydick-popups/modalBase should renders correctly VerticalButtonsView one button  1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "backgroundColor": "#F54D3D",
      "borderRadius": 18,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginTop": 36,
      "minHeight": 72,
      "opacity": 1,
    }
  }
>
  <Text
    allowFontScaling={true}
    style={
      Array [
        Object {
          "color": "#FFF",
          "fontFamily": "Inter-SemiBold",
          "fontSize": 27,
          "lineHeight": 33,
        },
        Array [
          Object {
            "color": "#FFF",
            "paddingHorizontal": 9,
          },
          undefined,
        ],
      ]
    }
  >
    text
  </Text>
</View>
`;

exports[`@npm/mobydick-popups/modalBase should renders correctly VerticalButtonsView secondary  1`] = `
Array [
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "rgba(43, 120, 238, 0.15)",
        "borderRadius": 18,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 36,
        "minHeight": 72,
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={true}
      style={
        Array [
          Object {
            "color": "#FFF",
            "fontFamily": "Inter-SemiBold",
            "fontSize": 27,
            "lineHeight": 33,
          },
          Array [
            Object {
              "color": "#2B78EE",
              "paddingHorizontal": 9,
            },
            undefined,
          ],
        ]
      }
    >
      text
    </Text>
  </View>,
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "rgba(43, 120, 238, 0.15)",
        "borderRadius": 18,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 18,
        "minHeight": 72,
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={true}
      style={
        Array [
          Object {
            "color": "#FFF",
            "fontFamily": "Inter-SemiBold",
            "fontSize": 27,
            "lineHeight": 33,
          },
          Array [
            Object {
              "color": "#2B78EE",
              "paddingHorizontal": 9,
            },
            undefined,
          ],
        ]
      }
    >
      text
    </Text>
  </View>,
]
`;
